analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs

```diff
@@ -75,11 +75,11 @@ protected override ProgramState PreProcess(ProgramState state, IPropertyReferenc
     {
         if (operation.Instance.TrackedSymbol() is { } symbol)
         {
-            if (!IsNullableHasValue(operation))
+            if (!IsNullableProperty(operation, "HasValue"))
             {
                 state = state.SetSymbolConstraint(symbol, ObjectConstraint.NotNull);
```

Comment:
Do you track work for the invocation `nullableInt.GetValueOrDefault()`? We need to do the opposite there: We don't learn  `ObjectConstraint.NotNull` on that invocation (same for `ToString()` and `GetHashCode()`, but not for `GetType()`).

Reply:
Added to the checklist